src/repository.c

```diff
@@ -236,17 +284,34 @@ static int load_workdir(git_repository *repo, git_config *config, git_buf *paren
 			&ce, config, "core.worktree", false)) < 0)
 		return error;
 
-	if (ce && ce->value) {
+	if (repo->is_worktree) {
+		char *gitlink = git_worktree__read_link(repo->gitdir, GIT_GITDIR_FILE);
+		if (!gitlink) {
+			error = -1;
+			goto cleanup;
+		}
+
+		git_buf_attach(&worktree, gitlink, 0);
+
+		if ((git_path_dirname_r(&worktree, worktree.ptr)) < 0
+			|| git_path_to_dir(&worktree) < 0) {
```

Comment:
One stylistic comment here:  the `||` should be hanging at the end of the line above it.  There are a bunch of occurrences of this that should be changed - I haven't noted each individually, but I can if that would be helpful.